packages/core/schematics/migrations/http-providers/utils.ts

```diff
@@ -177,6 +179,22 @@ function migrateDecorator(
     return;
   }
 
+  // HttpClient imported in component is actually a mistake
+  // Schematics will be no-op but add a TODO
+  const isComponent = decorator.name === 'Component';
+  if (isComponent && importedModules.client) {
+    const httpClientModuleIdentifier = importedModules.client;
+    const commentText = `\n// TODO: HttpClientModule should never be imported into a component directly.\n`;
```

Comment:
nit:
```suggestion
    const commentText = `\n// TODO: HttpClientModule should not be imported into a component directly.\n`;
```

I think it'd be great to make the message more actionable and explain what should be done instead, something like:

```
// TODO: `HttpClientModule` should not be imported into a component directly.
// Please refactor the code to add `provideHttpClient()` call to the provider list in the
// application bootstrap logic and remove the `HttpClientModule` import from this component.
```
The `provideHttpClient()` part of the message may also contain the necessary `with*` functions, based on the actual usage.